src/library/scala/collection/immutable/Queue.scala

```diff
@@ -100,6 +100,15 @@ class Queue[+A] protected(protected val in: List[A], protected val out: List[A])
     case _                               => super.:+(elem)(bf)
   }
 
+  override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
+    if (bf eq Queue.ReusableCBF) {
```

Comment:
@Ichoran When is it that `That =:= Self; val that: That = _; that.isInstanceOf[Self]` is not the case? Please provide an example.